kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportNamer.kt

```diff
@@ -549,23 +568,29 @@ internal class ObjCExportNamerImpl(
         }
     }
 
-    override fun getPropertyName(property: PropertyDescriptor): String = propertyNames.getOrPut(property) {
+    override fun getPropertyName(property: PropertyDescriptor): ObjCExportNamer.PropertyName {
         assert(mapper.isBaseProperty(property))
         assert(mapper.isObjCProperty(property))
-
-        StringBuilder().apply {
-            append(property.name.asString().toIdentifier())
-        }.mangledSequence {
-            append('_')
+        val objCName = property.getObjCName()
+        fun PropertyNameMapping.getOrPut(forSwift: Boolean) = getOrPut(property) {
+            StringBuilder().apply {
+                append(objCName.asIdentifier(forSwift))
+            }.mangledSequence {
+                append('_')
+            }
         }
+        return ObjCExportNamer.PropertyName(
+                swiftName = swiftPropertyNames.getOrPut(true),
+                objCName = objCPropertyNames.getOrPut(false)
+        )
     }
 
     override fun getObjectInstanceSelector(descriptor: ClassDescriptor): String {
         assert(descriptor.kind == ClassKind.OBJECT)
 
         return objectInstanceSelectors.getOrPut(descriptor) {
-            val name = descriptor.name.asString().replaceFirstChar(Char::lowercaseChar).toIdentifier().mangleIfSpecialFamily("get")
-
+            val name = descriptor.getObjCName().asString(false)
```

Comment:
Is this tested for `@ObjCName`-annotated `object` declaration?

Reply:
It is now :)